source/material_model/utilities.cc

```diff
@@ -514,58 +564,39 @@ namespace aspect
           enthalpy_values.reinit(n_temperature,n_pressure);
 
           unsigned int i = 0;
+          std::vector<double> previous_row_values(n_columns, 0.);
+
           while (!in.eof())
             {
-              double temp1,temp2;
-              double rho,alpha,cp,vp,vs,h;
-              in >> temp1 >> temp2;
-              in >> rho;
-              if (in.fail())
-                {
-                  in.clear();
-                  rho = density_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> alpha;
-              if (in.fail())
-                {
-                  in.clear();
-                  alpha = thermal_expansivity_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> cp;
-              if (in.fail())
-                {
-                  in.clear();
-                  cp = specific_heat_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> vp;
-              if (in.fail())
-                {
-                  in.clear();
-                  vp = vp_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> vs;
-              if (in.fail())
-                {
-                  in.clear();
-                  vs = vs_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> h;
-              if (in.fail())
+              std::vector<double> row_values(n_columns);
+
+              for (unsigned int n=0; n<n_columns; n++)
                 {
-                  in.clear();
-                  h = enthalpy_values[(i-1)%n_temperature][(i-1)/n_temperature];
+                  in >> row_values[n];
+                  // PerpleX P-T grids sometimes contain rows filled with NaNs
+                  // in the corners of the grid.
+                  // These P-T regions are unimportant, and we don't want ASPECT
+                  // to crash just because it sees a NaN.
+                  // Here we choose to replace invalid doubles
+                  // with the most recent valid double.
+                  if (in.fail())
```

Comment:
Is that what you really get? In other words, when you have NaN in the file, do you get `in.fail() == true` or is `row_values[n]` simply assigned the `nan` value?

Reply:
In this case, in.fail() == true and row_values[n] is assigned a 0.